app/window.py

```diff
@@ -873,21 +873,10 @@ def refresh(self):
   def setTextBuffer(self, textBuffer):
     app.log.info('setTextBuffer')
     #self.normalize()
-    if self.textBuffer is not None:
-      app.history.set(['files', self.textBuffer.fullPath, 'cursor'],
-          (self.textBuffer.penRow, self.textBuffer.penCol))
-    # TODO(dschuyler): Do we need to restore positions and selections here?
     self.controller.setTextBuffer(textBuffer)
     Window.setTextBuffer(self, textBuffer)
+    self.textBuffer.restoreUserHistory()
```

Comment:
I wondered why this was here rather than were the file is read (for symmetry with the file save code in this PR. Is it for the self.view.scroll* values? i.e. so that self has a self.view?
Consider not saving/restoring the scroll position. Try instead scrolling to the cursor position on open. If that works well, maybe this can be moved into the file loading code.